packages/sources/twosigma/src/endpoint/price.ts

```diff
@@ -0,0 +1,181 @@
+      await closed
+    }
+
+    subscriptions.new = subscriptions.desired
+    subscriptions.stale = []
+
+    await super.streamHandler(context, subscriptions)
+  }
+
+  override async sendMessages(
+    _context: EndpointContext<WebSocketEndpointTypes>,
+    subscribes: PriceEndpointParams[],
+    unsubscribes: unknown[],
+  ): Promise<void> {
+    // Send a single message containing the entire set of subscribed symbols rather than
+    // one message per symbol.
+
+    if (unsubscribes.length) {
+      // We explicitly set subscriptions.stale to empty.
+      logger.warn(`unexpected unsubscribes: ${JSON.stringify(unsubscribes)}`)
+    }
+
+    if (!subscribes.length) {
+      // No symbols to subscribe to
+      logger.debug(`nothing to subscribe to, skipping initial message`)
+      return
+    }
+
+    const payload: WebSocketRequest = {
+      api_key: process.env.WS_API_KEY || 'twosigma api key not set',
+      symbols: subscribes.map(buildSymbol).sort(),
+    }
+    this.wsConnection.send(this.serializeMessage(payload))
+  }
+}
+
+export const parseBaseQuote = (symbol: string): PriceEndpointParams | undefined => {
+  // "AAPL/USD" -> { base: AAPL, quote: USD }
+  const splits = symbol.split('/')
+  if (splits.length !== 2) {
+    return
+  }
+  const [base, quote] = splits
+  return { base, quote }
+}
+
+export const buildSymbol = ({ base, quote }: PriceEndpointParams): string => {
+  return `${base}/${quote}`
+}
+
+export const config = {
+  url: (context: EndpointContext<WebSocketEndpointTypes>): string => {
+    return context.adapterConfig.WS_API_ENDPOINT
+  },
+
+  handlers: {
+    message: (message: WebSocketMessage): ProviderResult<WebSocketEndpointTypes>[] | undefined => {
+      if (!message.timestamp || !message.symbol_price_dict) {
+        logger.debug(`Received invalid message: ${message}`)
+        return undefined
+      }
+
+      const results: ProviderResult<WebSocketEndpointTypes>[] = []
+      for (const symbol in message.symbol_price_dict) {
+        const priceData = message.symbol_price_dict[symbol]
+        const params = parseBaseQuote(symbol)
+        if (params === undefined) {
+          continue
+        }
+
+        results.push({
+          params,
+          response: {
+            result: priceData.price,
+            data: {
+              result: priceData.price,
+            },
+            timestamps: {
+              providerIndicatedTimeUnixMs: message.timestamp * 1000,
+            },
+          },
+        })
+      }
+
+      return results
+    },
+  },
+
+  // We don't want the 'subscribeMessage' and 'unsubscribeMessage' hooks because we override
+  // the 'sendMessages' on the transport to send a single message for the entire set of
+  // symbols. However, this empty object is necessary for the transport to actually call our
+  // overridden 'sendMessages'.
+  builders: {},
+}
+
+export const endpoint = new PriceEndpoint({
+  name: 'price',
+  aliases: ['stock'],
```

Comment:
Can you confirm this alias with data products?

Reply:
Yup, this was requested by them: https://chainlink-core.slack.com/archives/C02FM5ZP2TC/p1676674767521649